pkg/sql/tenant_deletion.go

```diff
@@ -83,9 +83,13 @@ func dropTenantInternal(
 	}
 
 	if settings.Version.IsActive(ctx, clusterversion.V23_1TenantNamesStateAndServiceMode) {
+		if ignoreServiceMode {
+			// Compatibility with CC serverless use of crdb_internal.destroy_tenant().
+			info.ServiceMode = mtinfopb.ServiceModeNone
+		}
 		// We can only check the service mode after upgrading to a version
 		// that supports the service mode column.
-		if !ignoreServiceMode && info.ServiceMode != mtinfopb.ServiceModeNone {
+		if info.ServiceMode != mtinfopb.ServiceModeNone {
```

Comment:
In case this confuses others as well: This fixes the bug not because of anything wrong with this if condition, but rather because UpdateTenantRecord (called below) performs additional validation, which is why we need to explicitly override the ServiceMode above.

Reply:
Is your feedback a request to add a comment?

Reply:
i added the comment nonetheless.

Reply:
I was mostly hoping to save Jay a few seconds of confusion in case he was reviewing it as well.  A comment may be nice, but I think it is fine to merge as-is.